codegen/src/main/java/software/amazon/awssdk/codegen/poet/batchmanager/BatchFunctionsClassSpec.java

```diff
@@ -109,11 +109,12 @@ private MethodSpec constructor() {
                              .collect(Collectors.toList());
     }
 
-    private Stream<MethodSpec> safeBatchFunctions(Map.Entry<String, BatchManagerMethods> batchFunctions) throws RuntimeException {
+    private Stream<MethodSpec> safeBatchFunctions(Map.Entry<String, BatchManagerMethods> batchFunctions)
+            throws IllegalArgumentException {
```

Comment:
`throws IllegalArgumentException` is not needed for unchecked exception

Reply:
Not throwing unchecked exception.